### PR TITLE
Worldmap: Add quest progress

### LINKED ISSUE #5589: Request for Quest Markers to Change When Quest is Completed on Minimap/Map
**Is your feature request related to a problem? Please describe.**
I'd like it if the minimap had some indication if a quest had been completed on the map, similar to how the agility shortcuts are crossed through if you can't use them.

**Describe the solution you'd like**
When a quest is completed, draw a red line through it signifying the quest is done.

**Describe alternatives you've considered**
Perhaps just hide the quest marker if the quest has been completed.

### PR DESCRIPTION
**Pull request by Rens-br**
Add option to show your quest progress on the world map.
If a quest is started it changes the quest icon and tool tip to show the progress (Started, Finished).

Screenshot_13
Screenshot_12

Closes #5589

### CODE REVIEW

**Comment by deathbeam** [comment]
needs rebase

**Comment by Hoten** [comment]
Oooh I really want this. @Rens-br are you able to rebase this? Would be happy to do it myself if you don't have the time.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapConfig.java [hunk 1]
```diff
@@ -33,175 +33,186 @@
 public interface WorldMapConfig extends Config
 {
 	@ConfigItem(
-		keyName = WorldMapPlugin.CONFIG_KEY_FAIRY_RING_TOOLTIPS,
-		name = "Show fairy ring codes in tooltip",
-		description = "Display the code for fairy rings in the icon tooltip",
-		position = 1
+			keyName = WorldMapPlugin.CONFIG_KEY_FAIRY_RING_TOOLTIPS,
```
Comment (Hoten) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapConfig.java]: accidental indents in this file

File: runelite-client/src/main/java/net/runelite/client/plugins/worldmap/QuestStartPoint.java [hunk 2]
```diff
@@ -30,10 +30,10 @@
 
 class QuestStartPoint extends WorldMapPoint
 {
-	QuestStartPoint(QuestStartLocation data, BufferedImage icon)
+	QuestStartPoint(QuestStartLocation data, BufferedImage icon, String toolTip)
 	{
 		super(data.getLocation(), icon);
 
-		setTooltip(data.getTooltip());
+		setTooltip("Quest start" + toolTip);
```
Comment (Hoten) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldmap/QuestStartPoint.java]: the tooltip is derived just from `data`, so could the switch on the quest state be moved to here? Then wouldn't need to add a third parameter to this constructor.

**Comment by Rens-br** [comment]
@Hoten I don't have time to rebase this at the moment. Feel free to do it yourself

**Comment by Adam-** [comment]
Closed in favor of #9086

**Pull request closed without merging**